inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -616,8 +659,40 @@ std::vector<PointSpeedPair> InLaneCruisingPlugin::maneuvers_to_points(const std:
       points_and_target_speeds.push_back(pair);
     }
   }
+    // Here we are limiting the trajectory length to the given length by maneuver end dist as opposed to the end of lanelets involved.
+    double starting_route_downtrack = wm_->routeTrackPos(points_and_target_speeds.front().point).downtrack; 
+    size_t i = 0;
+    size_t max_i = 0;
+    double dist_accumulator = starting_route_downtrack;
+    lanelet::BasicPoint2d prev_point;
+    for (auto point_speed_pair : points_and_target_speeds) {
+      auto current_point = point_speed_pair.point;
+      if (i == 0) {
+        prev_point = current_point;
+        continue;
+      }
+
+      double delta_d = lanelet::geometry::distance2d(prev_point, current_point);
+      dist_accumulator += delta_d;
+
+      double ending_downtrack = maneuvers.back().lane_following_maneuver.end_dist;
+      if(ending_downtrack + config_.buffer_ending_downtrack < wm_->getRouteEndTrackPos().downtrack){
+        ending_downtrack = ending_downtrack + config_.buffer_ending_downtrack;
+      } 
+
+      if (dist_accumulator > ending_downtrack) {
+        max_i = i;
+        break;
+      }
+      prev_point = current_point;
+      i++;
+    }
+    if (max_i == 0) {
+      max_i = points_and_target_speeds.size() - 1;
+    }
 
-  return points_and_target_speeds;
+    std::vector<PointSpeedPair> constrained_points(points_and_target_speeds.begin(), points_and_target_speeds.begin() + max_i);
```

Comment:
There needs to be a way to remove the extra points added just for supporting the spline fit after computing the curvature. This is what we discussed and you implemented in the car on 5/14 I believe.

Reply:
we are dropping the extra points around the line which has comment `//drop buffer points here`